src/main/java/app/classloaders/PluginLoader.java

```diff
@@ -0,0 +1,66 @@
+package app.classloaders;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.HashMap;
+
+/**
+ * Created by Ihar_Rubanovich on 2/23/2017.
+ */
+public class PluginLoader extends ClassLoader {
+    private HashMap<String, Class<?>> cache = new HashMap<String, Class<?>>();
+
+
+
+
+    @Override
+    public Class<?> loadClass(String name) throws ClassNotFoundException {
+        Class<?> loadClass = cache.get(name);
+
+        if (loadClass == null) {
+            loadClass = super.loadClass(name);
+        }
+        return loadClass;
+    }
+
+    public void loadPlugin(String pathToPlugin, String packageName, String strPluginFileName) throws IOException, ClassNotFoundException {
```

Comment:
`loadPlugin` method is the only way to load classes with this classloader;
default classloader api (`loadClass()`) will return only previously loaded classes (found in cache) or classes found by parent classloader (`super.loadClass()` will call parent and then call `findClass()` which throws `ClassNotFoundException` everytime)
you should override `findClass()` to load new classes via `loadClass()`